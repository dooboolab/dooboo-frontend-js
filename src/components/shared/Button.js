// @flow
import React, { Component } from 'react';
import styled from 'styled-components';
import { WhiteButton, DarkButton } from '../ui/Buttons';

type Props = {
  id?: string,
  white?: boolean,
  imgSrc?: any,
  srcset?: any,
  txt?: string,
  onClick: () => void,
};

type State = {

};

const Text = styled.span`
  font-size: 14px;
  color: rgb(128, 109, 216);
`;

const LogoImg = styled.img`
  position: absolute;
  left: 16px;
  height: 20px;
  width: 20px;
  object-fit: cover
`;

export class Button extends Component<Props, State> {
  static defaultProps: Props = {
    onClick: () => {},
  };

  render() {
    const { white, onClick, imgSrc, txt, srcset } = this.props;
    if (this.props.white) {
      return (
        <WhiteButton
          onClick={() => onClick()}
        >
          {
            this.props.imgSrc
              ? <LogoImg
                src={this.props.imgSrc}
                srcSet={this.props.srcset}
              />
              : null
          }
          <Text>{this.props.txt}</Text>
        </WhiteButton>
      );
    }
    return (
      <DarkButton
        onClick={() => onClick()}
      >
        {
          this.props.imgSrc
            ? <LogoImg
              src={this.props.imgSrc}
              srcSet={this.props.srcset}
            />
            : null
        }
        <Text>{this.props.txt}</Text>
      </DarkButton>
    );
  }
}

export default Button;
