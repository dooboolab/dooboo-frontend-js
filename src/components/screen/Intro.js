// @flow
import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';
import styled from 'styled-components';

import Button from '../shared/Button';

import { inject, observer } from 'mobx-react';
import { media } from '../../theme';
import Store from '../../stores/appStore';

const Container = styled.div`
  display: flex;
  height: 100vh;
  width: 100vw;
  align-self: stretch;
  overflow: scroll;
  background: linear-gradient(to bottom right, rgb(12, 157, 197), rgb(201, 109, 216));

  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  overflow: hidden;
`;

const Box = styled.div`
  position: absolute;
  width: 85vw;
  bottom: 40px;

  ${media.mobile`
    width: 50vw;
  `}
`;

type Props = {
  store: Store;
}

type State = {

};

@inject('store') @observer
class Intro extends Component<Props, State> {
  render() {
    const { getString } = this.props.store.locale;
    return (
      <Container>
        <Box>
          <Button
            id='btn'
            onClick={() => this.onClick()}
            white={true}
            txt={getString('BUTTON')}
          />
        </Box>
      </Container>
    );
  }

  onClick = () => {
    alert('clicked');
  }
}

export default Intro;
