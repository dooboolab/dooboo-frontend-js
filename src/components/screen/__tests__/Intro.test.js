import React from 'react';
import renderer from 'react-test-renderer';

import Intro from '../Intro';
import Button from '../../shared/Button';
import Store from '../../../stores/appStore';

const props = {
  store: new Store(),
  navigation: {
    navigate: jest.fn(),
  },
  history: {
    push: jest.fn(),
  },
};

// test for the container page in dom
describe('Intro page DOM rendering test', () => {
  let tree;
  const component = <Intro { ...props } />;

  it('component and snapshot matches', () => {
    tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

// describe('Interaction', () => {
//   let rendered;
//   let instance;
//   const component = <Intro { ...props } />;

//   beforeAll(() => {
//     rendered = renderer.create(component);
//     instance = rendered.root;

//     Object.defineProperty(window.alert, 'window.alert', {
//       configurable: true,
//     });
//     window.alert = jest.fn();
//   });

//   it('Simulate onClick', () => {
//     // const spy = jest.spyOn(rendered.getInstance(), 'onClick');
//     const button = instance.findByType(Button);
//     button.props.onClick();
//     expect(window.alert).toBeCalled();
//   });
// });

describe('Interaction', () => {
  let rendered: TestRenderer.ReactTestRenderer;
  let root: TestRenderer.ReactTestRenderer.root;
  const component = <Intro { ...props } />;

  beforeAll(() => {
    rendered = renderer.create(component);
    root = rendered.root;
  });

  it('Simulate onPress', () => {
    jest.useFakeTimers();
    const spy = jest.spyOn(rendered.getInstance().wrappedInstance, 'onLogin');
    const buttons = root.findAllByType(Button);
    root.instance.wrappedInstance.onLogin(); // == buttons[0].props.onPress();
    expect(setTimeout).toHaveBeenCalledTimes(1);
    expect(root.instance.wrappedInstance.state.isLoggingIn).toEqual(true);

    jest.runAllTimers();
    expect(root.instance.wrappedInstance.state.isLoggingIn).toEqual(false);
    expect(props.store.user.displayName).toEqual('dooboolab');
    expect(props.store.user.age).toEqual(30);
    expect(props.store.user.job).toEqual('developer');
    expect(spy).toBeCalled();
    buttons[1].props.onPress();
    expect(props.history.push).toBeCalledWith({
      pathname: '/404',
      state: {},
    });

    buttons[0].props.onPress();
  });
});

