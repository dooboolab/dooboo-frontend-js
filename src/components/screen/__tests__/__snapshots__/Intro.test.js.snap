// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intro page DOM rendering test component and snapshot matches 1`] = `
<div
  className="sc-ifAKCX fgUhTV"
>
  <div
    className="sc-EHOje fBGEpx"
  >
    <button
      className="sc-bwzfXH idXafH"
      onClick={[Function]}
    >
      <span
        className="sc-htpNat gUfFlb"
      >
        Button
      </span>
    </button>
  </div>
</div>
`;
